nodes/object/object_visibility_output.py

```diff
@@ -0,0 +1,160 @@
+
+    def create(self):
+        self.inputs.new("an_ObjectSocket", "Object", "object").defaultDrawType = "PROPERTY_ONLY"
+        self.inputs.new("an_BooleanSocket", "Hide", "hide").value = False
+        self.inputs.new("an_BooleanSocket", "Hide Select", "hideSelect").value = False
+        self.inputs.new("an_BooleanSocket", "Hide Render", "hideRender").value = False
+        self.inputs.new("an_BooleanSocket", "Show Name", "showName").value = False
+        self.inputs.new("an_BooleanSocket", "Show Axis", "showAxis").value = False
+        self.inputs.new("an_BooleanSocket", "Show Xray", "showXray").value = False
+        self.outputs.new("an_ObjectSocket", "Object", "object")
+        self.updateSocketVisibility()
+        
+    def draw(self, layout):
+        rrow = layout.row()
+        
+        col = rrow.column()
+        row = col.row(align = True)
+        row.alignment = 'LEFT'
+        #row.label("Visibility")
+        row.prop(self, "useView", text = "", icon = "RESTRICT_VIEW_OFF")
+        row.prop(self, "useSelect", text = "", icon = "RESTRICT_SELECT_OFF")
+        row.prop(self, "useRender", text = "", icon = "RESTRICT_RENDER_OFF")
+        
+        col = rrow.column()
+        row = col.row(align = True)
+        row.alignment = 'RIGHT'
+        #row.label("Show")
+        row.prop(self, "useName", text = "", icon = "SORTALPHA")    # SYNTAX_ON / SYNTAX_OFF
+        row.prop(self, "useAxis", text = "", icon = "MANIPUL")    
+        row.prop(self, "useXray", text = "", icon = "ROTACTIVE")    # MOD_DECIM / META_BALL
+    
+    def drawAdvanced(self, layout):
+        layout.operator("wm.call_menu", text = "Node Info / Help", icon = "INFO").name = "an.show_obj_vis_help"
+
+    def updateSocketVisibility(self):
+        self.inputs["Hide"].hide = not (self.useView)
+        self.inputs["Hide Select"].hide = not (self.useSelect)
+        self.inputs["Hide Render"].hide = not (self.useRender)
+        self.inputs["Show Name"].hide = not (self.useName)
+        self.inputs["Show Axis"].hide = not (self.useAxis)
+        self.inputs["Show Xray"].hide = not (self.useXray)
+        
+    def getExecutionCode(self):
+        lines = []
+        lines.append("if object is not None:")
+        
+        if self.useView: lines.append("    object.hide = hide")
+        if self.useSelect: lines.append("    object.hide_select = hideSelect")
+        if self.useRender: lines.append("    object.hide_render = hideRender")
+        
+        if self.useName: lines.append("    object.show_name = showName")
+        if self.useAxis: lines.append("    object.show_axis = showAxis")
+        if self.useXray: lines.append("    object.show_x_ray = showXray")
+        
+        if not any((self.useView, self.useSelect, self.useRender, self.useName, self.useAxis, self.useXray)): 
+            lines = []
+            
+        #lines.append("object = object")
+        return lines
+
+
+class ShowHelp(bpy.types.Menu):
```

Comment:
Until we have a better system for help menus we leave this as it is :)
only the bl_idname should be "an.show_object_visibilty_help"